### PR TITLE
[CSLD-126] Describe new topology, WIP

### PR DESCRIPTION
**Pull request by denisshevchenko**

### CODE REVIEW

File: _docs/cardano/2017-02-07-topology.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+---
+<!-- Reviewed at dec0d911d6c4beb8e708ed4076f832ff871f6125 -->
+
+# Cardano SL Network Topology
+
+This is an overview of Cardano SL network topology.
+
+## Nodes Groups
+
+Cardano SL is a distributed network: it consists of nodes. But currently all these nodes are divided into
```
Comment (georgeee) [inline _docs/cardano/2017-02-07-topology.md]: Should be described differently.

> Though Cardano SL is designed and implemented as distributed network, for purpose of DDOS protection additional facilities were added to CSL implementation and topology.

File: _docs/cardano/2017-02-07-topology.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+1. Core
+2. Relay
+3. Edge
+
+Let's describe each of these groups.
+
+### Core Node
+
+You can think of core nodes as of Cardano SL core.
```
Comment (georgeee) [inline _docs/cardano/2017-02-07-topology.md]: No
This phrase says nothing

File: _docs/cardano/2017-02-07-topology.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+These nodes are the most important ones: if they are alive, the blockchain cannot be compromised. So core
```
Comment (georgeee) [inline _docs/cardano/2017-02-07-topology.md]: > These nodes are the most important ones: if they are alive, the blockchain cannot be compromised

Why not just say that they are only nodes creating blocks (at least in Byron) and hence their operation is essential for network to operate

File: _docs/cardano/2017-02-07-topology.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+nodes are isolated: they are fully under the control of the IOHK and cannot be directly accessible from
```
Comment (georgeee) [inline _docs/cardano/2017-02-07-topology.md]: > they are fully under the control of the IOHK and cannot be directly accessible

These details should not be here, they relate to description of bootstrap era (or Byron release).

What we need to say here they're isolated and put in perimeter of relay nodes, and only relay nodes are allowed to communicate to core nodes

File: _docs/cardano/2017-02-07-topology.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+the public internet. The reason of such an isolation is a DoS-protection: if nodes cannot be directly
+accessible - they cannot be directly attacked.
```
Comment (georgeee) [inline _docs/cardano/2017-02-07-topology.md]: My idea of how this paragraph can be party reformulated

> As is described in [...] for Byron we will have stake effectively locked on federation of core nodes. Only these few nodes will be creating blocks over this period. This is essential for network core nodes operate well and continue maintaining blockchain. And we bring additional level of security around this set of core nodes: we put them inside a perimeter of relay nodes.

Comment (georgeee) [inline _docs/cardano/2017-02-07-topology.md]: > they cannot be directly attacked

remove this and replace with something like we reduce probability from them being attacked

File: _docs/cardano/2017-02-07-topology.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+You can think of relay nodes as of proxy between core nodes and public internet. 
+
+Since relay nodes are not isolated they can be attacked but they are in principle stateless and don't have
+any stake, so they can be moved or their number can be increased. If relays are taken down, this does deny
+service, but the integrity of the core nodes (and the blockchain) should not be compromised.
+
+Since relay nodes don't have any stake, they cannot be slot leaders. Moreover, they cannot create currency
+transactions as well.
+
+Relay nodes are fully under the control of the IOHK and two our partners: Cardano Foundation and SGG.
```
Comment (georgeee) [inline _docs/cardano/2017-02-07-topology.md]: Not IOHK, but federated committee of initial stakeholders!

**Review by georgeee — CHANGES_REQUESTED** [review]

**Review by staenrey — APPROVED** [review]
grammar checked

**Pull request merged**